src/textord/edgblob.h

```diff
@@ -46,12 +46,12 @@ class OL_BUCKETS
     C_OUTLINE_LIST *start_scan() {
       for (index = 0; buckets[index].empty () && index < bxdim * bydim - 1;
         index++);
-      return &buckets[index];
+      return &buckets[index++];  //index value has not changed
```

Comment:
The old code works because the for loop is left with `buckets[index]` being not empty, but being emptied later before calling the `scan_next` method. That also applies to the results of `scan_next`. See calling code in function `empty_buckets`.

Increasing `index` like it is done here would be needed in other contexts (like the test case in the descriptions) and could also improve the performance a little bit, but needs special handling for the corner case where `index` reaches the upper limit.